File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string firstStr = strs[0];
        string lastStr = strs[strs.size()-1];

        int n = min(firstStr.size(), lastStr.size());
        string ans = "";
        for(int i=0; i < n; i++) {
            if(firstStr[i] == lastStr[i]) ans+=firstStr[i];
            else break;
        }

        return ans;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {

        StringBuilder sb = new StringBuilder();
        int i = 0;

        while (true) {
            boolean flag = false;
            if (i >= strs[0].length())
                break;
            for (String str : strs) {
                if (i >= str.length()) {
                    flag = true;
                    break;
                }
                if (strs[0].charAt(i) != str.charAt(i)) {
                    flag = true;
                    break;
                }

            }

            if (flag)
                break;
            sb.append(strs[0].charAt(i));
            i++;
        }

        return sb.toString();
    }
}